### DISCUSSION TITLE
How can users include dynamic labels on their creations?

### DISCUSSION THREAD

**Discussion by hooperbloob** [FreeCAD/FreeCAD#15475] [Discussions that will turn in to issues but need refining]
If you purchase plastic plumbing fittings at the hardware store, chances are they'll be imprinted with info describing their critical dimensions. 

For my project, as I iterate through a number of versions, I'd like to incorporate text on them that can be drawn from a spreadsheet or other external means to keep track of them.   Unfortunately the shapestrings creation dialog doesn't support expression entry.  Is there another way to add dynamic labels to a body?

**Comment by krushia** [comment]
You can use an expression for the ShapeString text (property `String`) in the Property view.

**Reply by hooperbloob** [reply]
Thanks, but I'm not seeing the blue dot in the text field that brings up the expression editor:

<img width="764" alt="Screenshot 2024-07-18 at 9 02 05 AM" src="https://github.com/user-attachments/assets/4cae3f4c-01b8-4b5e-928f-33b4439d388e">

It just shows 'Default' as the initial text. Is there another editor that can do this?
